### PR TITLE
Kyberdao UI

### PR DESCRIPTION
**Pull request by XiaoYhun**

### CODE REVIEW

**Review by hieuvo — APPROVED** [review]

**Review by Linhhnk — DISMISSED** [review]

File: package.json [hunk 1]
```diff
@@ -66,6 +66,8 @@
     "@web3-react/walletconnect-connector": "^7.0.2-alpha.0",
     "@web3-react/walletlink-connector": "^6.2.14",
     "aos": "^2.3.4",
+    "env-cmd": "^10.1.0",
+    "eslint-plugin-unused-imports": "^2.0.0",
```
Comment (viet-nv) [inline package.json:70]: why do we need these dependencies?

**Review by viet-nv — APPROVED** [review]

File: src/components/Header/index.tsx [hunk 2]
```diff
@@ -402,59 +409,48 @@ export default function Header() {
               <Trans>Swap</Trans>
               <DropdownIcon />
             </Flex>
-
-            <Dropdown id={TutorialIds.BRIDGE_LINKS}>
```
Comment (nguyenhoaidanh) [inline src/components/Header/index.tsx:406]: maybe conflict when rebase/merge
help me update to `<Dropdown id=  id={TutorialIds.BRIDGE_LINKS}>` and remove div `<div id={TutorialIds.BRIDGE_LINKS}>`

Comment (namgold) [reply src/components/Header/index.tsx:406]: `id= id=` mistake?

File: src/pages/KyberDAO/StakeKNC/YourTransactionsModal.tsx [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+  const pageSize = isMobile ? 5 : 10
+  const { stakerActions } = useStakingInfo()
+  const formattedActions: (StakerAction & { hashText: string; description: ReactNode })[] = useMemo(
+    () =>
+      stakerActions?.slice((page - 1) * pageSize, page * pageSize)?.map((action: StakerAction) => {
+        return {
+          ...action,
+          hashText: action.tx_hash.slice(0, 6) + '...' + action.tx_hash.slice(-4),
+          type:
+            action.type === ActionType.VoteEmitted
```
Comment (nguyenhoaidanh) [inline src/pages/KyberDAO/StakeKNC/YourTransactionsModal.tsx:119]:
```
const map = {
  [ActionType.VoteEmitted]: 'Vote',
  ...
}
```

File: src/pages/KyberDAO/StakeKNC/YourTransactionsModal.tsx [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+                      Power
+                    </Text>
+                  </>
+                )
+              }
+              case ActionType.Deposit: {
+                const amount = action.meta?.amount ?? 0
+                return (
+                  <>
+                    {amount > 0 && amount < 0.001 ? '<0.001' : amount?.toLocaleString() + ' KNC'}
```
Comment (nguyenhoaidanh) [inline src/pages/KyberDAO/StakeKNC/YourTransactionsModal.tsx:150]: duplicate, make it to utils func

File: src/pages/KyberDAO/Vote/SelectProposalStatus.tsx [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+            return (
+              <DropdownItem
+                key={s}
+                active={s === status}
+                onClick={() => {
+                  setShow(false)
+                  setStatus?.(s)
+                }}
+              >
+                <Trans>{s}</Trans>
```
Comment (nguyenhoaidanh) [inline src/pages/KyberDAO/Vote/SelectProposalStatus.tsx:116]: remove Trans and update t`xxx`
```
export enum ProposalStatus {
  Pending = 'Pending',
  Active = 'Active',
  Approved = 'Approved',
  Executed = 'Executed',
  Failed = 'Failed',
  Canceled = 'Canceled',
}
```

File: src/pages/KyberDAO/Vote/VoteConfirmModal.tsx [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+            <Text fontSize={20}>
+              <Trans>Vote</Trans>
+            </Text>
+          </AutoRow>
+          <Flex sx={{ cursor: 'pointer' }} role="button" onClick={toggle}>
+            <X onClick={toggle} size={20} color={theme.subText} />
+          </Flex>
+        </RowBetween>
+        <TextWrapper fontSize={16} lineHeight="24px" color={theme.subText}>
+          You are voting for <b>{options}</b> on <b>{title}</b> with your KIP voting power
```
Comment (nguyenhoaidanh) [inline src/pages/KyberDAO/Vote/VoteConfirmModal.tsx:51]: trans

File: src/pages/KyberDAO/Vote/index.tsx [hunk 7]
```diff
@@ -0,0 +1,464 @@
...
+    [Math.floor((((seconds % 31536000) % 86400) % 3600) / 60), 'm'],
+  ]
+
+  let returntext = ''
+  for (let i = 0, max = levels.length; i < max; i++) {
+    if (levels[i][0] === 0) continue
+    returntext +=
+      ' ' +
+      levels[i][0] +
+      '' +
```
Comment (nguyenhoaidanh) [inline src/pages/KyberDAO/Vote/index.tsx:109]: may redudant

File: src/pages/KyberDAO/Vote/SelectProposalStatus.tsx (hunk 5 above)
Comment (XiaoYhun) [reply src/pages/KyberDAO/Vote/SelectProposalStatus.tsx:116]: cannot do this as the text is inside enum which cannot be computed. Keep it as English is fine I think

File: src/pages/KyberDAO/Vote/SelectProposalStatus.tsx [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+  status?: string
+  setStatus?: (s: string) => void
+}) {
+  const theme = useTheme()
+  const [show, setShow] = useState(false)
+  const ref = useRef()
+  useOnClickOutside(ref, () => setShow(false))
+  return (
+    <>
+      <Wrapper ref={ref as any}>
```
Comment (namgold) [inline src/pages/KyberDAO/Vote/SelectProposalStatus.tsx:90]:
```suggestion
  const ref = useRef<HTMLDivElement>(null)
  useOnClickOutside(ref, () => setShow(false))
  return (
    <>
      <Wrapper ref={ref}>
```

File: src/components/Header/groups/KyberDaoGroup.tsx [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+      <NavGroup
+        isActive={isActive}
+        anchor={
+          <DropdownTextAnchor active={isActive}>
+            <Trans>KyberDAO</Trans>
+          </DropdownTextAnchor>
+        }
+        dropdownContent={
+          <Column>
+            <StyledNavLink id={`kyberdao-stake-knc`} to={'/kyberdao/stake-knc'} style={{ gap: '4px' }}>
```
Comment (nguyenhoaidanh) [inline src/components/Header/groups/KyberDaoGroup.tsx:36]: use APP_PATHS

**Comment by nguyenhoaidanh** [comment]
lgtm, @namgold will approve

File: src/pages/KyberDAO/Vote/ProposalItem/VoteInformation.tsx [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+        </Text>
+        <Text color={theme.text}>{Math.floor(proposal.vote_stats.total_vote_count).toLocaleString()}</Text>
+      </InfoRow>
+      <InfoRow>
+        <Text color={theme.subText}>
+          <Trans>Your KIP Voting Power</Trans>{' '}
+          <InfoHelper
+            placement="top"
+            text="Your KIP Voting Power is calculated by
+            [Your Staked KNC] / [Total Voted KNC in this KIP] * 100%"
```
Comment (namgold) [inline src/pages/KyberDAO/Vote/ProposalItem/VoteInformation.tsx:83]:
```suggestion
            text={t`Your KIP Voting Power is calculated by
            [Your Staked KNC] / [Total Voted KNC in this KIP] * 100%`}
```

File: src/pages/KyberDAO/StakeKNC/index.tsx [hunk 11]
```diff
@@ -0,0 +1,224 @@
...
+            <CardInfo>
+              <Text fontSize={20} lineHeight="24px" fontWeight={500} color={theme.text}>
+                <Trans>Stake + Vote</Trans>
+              </Text>
+              <Text fontSize={12} lineHeight="16px" fontWeight={500} color={theme.subText}>
+                <Trans>The more you stake and vote, the more KNC you will earn. </Trans>
+              </Text>
+            </CardInfo>
+            <ButtonPrimary onClick={() => navigate('/kyberdao/vote')} width="120px" height="44px">
+              Vote
```
Comment (namgold) [inline src/pages/KyberDAO/StakeKNC/index.tsx:166]:
```suggestion
              <Trans>Vote</Trans>
```

File: src/pages/KyberDAO/Vote/ProposalItem/index.tsx [hunk 12]
```diff
@@ -0,0 +1,418 @@
...
+      </OptionsWrapper>
+    )
+  }, [proposal, selectedOptions, votedOfCurrentProposal?.options, handleOptionClick, isActive])
+
+  return (
+    <ProposalItemWrapper>
+      <ProposalHeader>
+        <RowBetween onClick={() => setShow(s => !s)}>
+          <Text>
+            <Trans>{proposal.title}</Trans>
```
Comment (namgold) [inline src/pages/KyberDAO/Vote/ProposalItem/index.tsx:325]:
```suggestion
            {proposal.title}
```

**Review by namgold — APPROVED** [review]

**Pull request merged**